anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -483,201 +482,240 @@ def image(dbsession, request_inputs, bodycontent=None):
                     raise Exception("image not found in DB")
 
         elif method == "POST":
-            timer = time.time()
+            return_object, httpcode = image_post(
```

Comment:
if we're going this far, might as well split the handlers for get and post into two different methods, create a common pre-processing method, and update the calls in anchore_engine/services/catalog/api/controllers/default_controller.py. will be easier to read. having the POST handler tacked on to the end of the images() method feels like an afterthought

Reply:
What would you think about refactoring this in a subsequent change? Wanted to minimize the impact here as there _is_ some pre-processing that is common for the two methods for now

Reply:
That's fine w me.